src/onnxflow/justbuildit/export.py

```diff
@@ -0,0 +1,614 @@
+                    raise exp.StageError(msg)
+
+        if opset < build.DEFAULT_ONNX_OPSET and opset >= build.MINIMUM_ONNX_OPSET:
+            print(
+                " \n The received model has an opset {opset}. Though this opset is supported \
+                we recommend upgrading the model to opset {build.MINIMUM_ONNX_OPSET}"
+            )
+        elif opset < build.MINIMUM_ONNX_OPSET:
+            msg = f"""
+            The received model has an opset {opset}. Opset < 11 is not supported. Please
+            try upgrading the model to opset 13.
+            More information may be available in the log file at **{self.logfile_path}**
+            """
+            raise exp.StageError(msg)
+
+        shutil.copy(state.model, state.base_onnx_file)
+
+        # Save inputs and convert to fp16 and int32 (no int64 nor float32/64)
+        to_downcast = False if state.quantization_samples else True
+        tensor_helpers.save_inputs(
+            [state.inputs], state.original_inputs_file, downcast=to_downcast
+        )
+
+        # Check the if the base mode has been exported successfully
+        success_msg = "\tSuccess receiving ONNX Model"
+        fail_msg = "\tFailed receiving ONNX Model"
+        state.info.base_onnx_exported = _check_model(
+            state.base_onnx_file, success_msg, fail_msg
+        )
+
+        if state.info.base_onnx_exported:
+            state.intermediate_results = [state.base_onnx_file]
+        else:
+            msg = f"""
+            Unable to process ONNX Model. We recommend that you verify the source of the model.
+            Any optimizations performed on the model could result in an error.
+            More information may be available in the log file at **{self.logfile_path}**
+            """
+            raise exp.StageError(msg)
+
+        return state
+
+
+class ExportPytorchModel(stage.Stage):
+    """
+    Stage that takes a PyTorch model instance, in state.model, and
+    exports it to an ONNX file.
+
+    Expected inputs:
+     - state.model is a torch.nn.Module or torch.jit.ScriptModule
+     - state.inputs is a dict that represents valid kwargs to the forward
+        function of state.model
+
+    Outputs:
+     - A *-base.onnx file that implements state.model given state.inputs
+    """
+
+    def __init__(self):
+        super().__init__(
+            unique_name="export_pytorch",
+            monitor_message="Exporting PyTorch to ONNX",
+        )
+
+    def fire(self, state: build.State):
+        if not isinstance(state.model, (torch.nn.Module, torch.jit.ScriptModule)):
+            msg = f"""
+            The current stage (ExportPytorchModel) is only compatible with
+            models of type torch.nn.Module or torch.jit.ScriptModule, however
+            the stage received a model of type {type(state.model)}.
+            """
+            raise exp.StageError(msg)
+
+        # TODO: check that state.inputs is valid
+        # https://git.groq.io/code/Groq/-/issues/13947
```

Comment:
* Remove link to GroqFlow issue
* Replace with MLAgility issue
* Cross reference: Mention MLAgility issue in the GroqFlow issue